lockfile/src/lib.rs

```diff
@@ -237,7 +237,8 @@ pub struct ThirdPartyVersion {
 ///
 /// The file does not need to exist.
 pub fn get_path_format<P: AsRef<Path>>(path: P) -> Option<LockfileFormat> {
-    LockfileFormat::iter().find(|f| f.parser().is_path_lockfile(path.as_ref()))
+    LockfileFormat::iter()
+        .find(|f| f != &LockfileFormat::GoMod && f.parser().is_path_lockfile(path.as_ref()))
 }
```

Comment:
Is it necessary to change this? It should still be possible to detect go files on their own.

I'd assume this was changed because of the GitHub app?

Reply:
Yes this is for the GitHub app. However maybe the change to prefer go.sum first should be done in the app?

Reply:
@kylewillmon @matt-phylum thoughts?

Reply:
We need some way to prefer the correct file type or iteratively work through them. It's okay if we detect that both files may be lockfiles we're interested in, grab both, and then decide that we only needed one of them, but right now we detect which file we _are_ interested in, grab that file, and error if we can't use it, and this code I think is supposed to make the current code work.

IIRC it is possible, although undocumented, for users to specify lockfiles instead of us autodetecting, which may be another option if we have a small number of users for which they need to use the older scheme.

Reply:
If the autodetection had some way to specify that these two files represent the same dependencies (as opposed to go.mod in one folder and go.lock in a different folder, or go.mod and package-lock.json in the same folder) as options for a single dependency source, we could support trying and failing individual files.

Reply:
Wouldn't CLI have the same problem? Without this code, if it detects both go.mod and go.sum, does it have a way to pick the correct file or does it use both unless told otherwise?

Reply:
> Wouldn't CLI have the same problem? Without this code, if it detects both go.mod and go.sum, does it have a way to pick the correct file or does it use both unless told otherwise?

Yes, I removed it from being detected in the `find_at` function. You can still target it manually via `phylum analyze -t gomod go.mod`.

Reply:
I removed this change, however we'll need a new issue for the GitHub app.